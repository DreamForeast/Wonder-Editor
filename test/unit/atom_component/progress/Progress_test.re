open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("Progress component", () => {
    let sandbox = getSandboxDefaultVal();

    let _buildProgressComponent = () =>
      ReactTestRenderer.create(<Progress />);

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test progress component", () => {
      test("test snapshot", () =>
        _buildProgressComponent() |> ReactTestTool.createSnapshotAndMatch
      );

      describe("test didMount", () => {
        test("bind change percent event", () => {
          let send = ProgressTool.didMount(sandbox);

          ProgressUtils.changePercent(10)
          |> StateLogicService.getAndSetEngineState;

          send |> expect |> toCalledWith([|Progress.ChangePercent(10)|]);
        });
        test("bind show event", () => {
          let send = ProgressTool.didMount(sandbox);

          ProgressUtils.show |> StateLogicService.getAndSetEngineState;

          send |> expect |> toCalledWith([|Progress.Show|]);
        });
        test("bind hide event", () => {
          let send = ProgressTool.didMount(sandbox);

          ProgressUtils.show |> StateLogicService.getAndSetEngineState;

          send |> expect |> toCalledWith([|Progress.Show|]);
        });
      });

      describe("test willUnmount", () =>
        describe("off custom global event", () =>
          test("test off hide event", () => {
            let send = ProgressTool.didMount(sandbox);
            ProgressTool.willUnmount();

            ProgressUtils.show |> StateLogicService.getAndSetEngineState;

            send |> expect |> not_ |> toCalledWith([|Progress.Show|]);
          })
        )
      );
    });
  });