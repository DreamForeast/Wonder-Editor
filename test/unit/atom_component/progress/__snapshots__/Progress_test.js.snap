// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component test progress component test snapshot 1`] = `
<article
  className="wonder-progress"
  id="wonder-progress"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="progress-content"
  >
    <div
      className="content-percent"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      0%
    </div>
  </div>
  <div
    className="progress-bg"
  />
</article>
`;
